dev/tools/create_overlays.sh

```diff
@@ -0,0 +1,78 @@
+#!/usr/bin/env bash
+
+# TODO:
+#
+# - Check if the branch already exists in the remote => checkout
+# - Better error handling
+# - Just checkout, don't build
+# - Rebase functionality
+#
+
+set -x
+set -e
+set -o pipefail
+
+# setup_contrib_git("_build_ci/fiat", "https://github.com/ejgallego/fiat-core.git")
+setup_contrib_git() {
+
+    local _DIR=$1
+    local _GITURL=$2
+
+    ( cd $_DIR
+      git checkout -b $OVERLAY_BRANCH         || true # allow the branch to exist already
+      git remote add $DEVELOPER_NAME $_GITURL || true # allow the remote to exist already
+    )
+
+}
+
+if [ $# -lt 3 ]; then
+  echo "usage: $0 github_username pr_number contrib1 ... contribN"
+  exit 1
+fi
+
+set +x
+. dev/ci/ci-basic-overlay.sh
+set -x
+
+DEVELOPER_NAME=$1
+shift
+PR_NUMBER=$1
+shift
+OVERLAY_BRANCH=$(git rev-parse --abbrev-ref HEAD)
+OVERLAY_FILE=$(mktemp overlay-XXXX)
```

Comment:
What's the point of doing this instead of `OVERLAY_FILE=dev/ci/user-overlays/...`?

Reply:
The script can fail for several reasons and then it would leave a corrupt file in `dev/ci/user-overlays`; this corrupt file would be then included in the next `make` call creating havoc.